internal/server/mcp.go

```diff
@@ -164,6 +202,19 @@ func mcpHandler(s *Server, w http.ResponseWriter, r *http.Request) {
 		res = newJSONRPCError(baseMessage.Id, mcp.METHOD_NOT_FOUND, fmt.Sprintf("invalid method %s", baseMessage.Method), nil)
 	}
 
+	// retrieve sseSession
+	sessionId := r.URL.Query().Get("sessionId")
+	session, ok := s.sseSessions[sessionId]
+	if !ok {
+		fmt.Printf("sse session not available\n")
+	} else {
+		// send response via sse if sse session is available
+		eventData, _ := json.Marshal(res)
+		fmt.Fprintf(session.writer, "event: message\ndata: %s\n\n", eventData)
+		session.flusher.Flush()
```

Comment:
This introduces a potential race condition because it has multiple goroutines writing to the same pipe. 

We'll need to restructure so that a single responses are written at once and they are passed to a single point via a channel.